homeassistant/components/switch_as_x/__init__.py

```diff
@@ -101,3 +101,20 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return await hass.config_entries.async_unload_platforms(
         entry, (entry.options[CONF_TARGET_DOMAIN],)
     )
+
+
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Unload a config entry."""
+    # Unhide the wrapped entry if registered
+    registry = er.async_get(hass)
+    try:
+        entity_id = er.async_validate_entity_id(registry, entry.options[CONF_ENTITY_ID])
```

Comment:
Why validate ? Doesn't the below check if an entry is returned do the same?

Reply:
The function also resolves `uuid` to `entity_id`. I think the name `async_validate_entity_id` is unclear. It used to be `async_resolve_entity_id`, but we changed it because the function raises if the uuid is missing.